src/ManageCourses.Api/Middleware/BearerTokenHandler.cs

```diff
@@ -44,57 +44,56 @@ protected override async Task<AuthenticateResult> HandleAuthenticateAsync()
                 return AuthenticateResult.NoResult();
             }
 
+            SubjectAndEmail subjectAndEmail = null;
             try
             {
-                var userDetails = GetJsonUserDetailsFromDatabase(accessToken) ?? GetJsonUserDetailsFromOauth(accessToken);
-                
-                try
+                subjectAndEmail = GetSubjectAndEmailFromDatabase(accessToken);
+                if (subjectAndEmail == null)
                 {
+                    var userDetails = GetJsonUserDetailsFromOauth(accessToken);
+                    subjectAndEmail = new SubjectAndEmail(userDetails.Subject, userDetails.Email);
                     await _userService.UserSignedInAsync(accessToken, userDetails);
```

Comment:
you'll lose the login log by moving the call to `UserSignedInAsync()` into the `if`

https://github.com/DFE-Digital/manage-courses-api/blob/ae1424fff491e74b8c7f002917f4da859f594cfe/src/ManageCourses.Api/Services/Users/UserService.cs#L46

reject

Reply:
fixed :+1: